### ISSUE TITLE
several issues with new release channels

### ISSUE THREAD

**Issue by tooomm** [Cockatrice/Cockatrice#2465]
<b>OS:</b>
Win 10 64bit

<b>Cockatrice version:</b>
02bb226 (2017-03-12)

___
<br>

Feature initially introduced in https://github.com/Cockatrice/Cockatrice/pull/2362

## Issue 1:
Seems to be Windows only, since it seems to work for @ZeldaZach on macOS.
I realized that it's not possible to switch update channels. If you change the channel in settings and reopen the settings window it'll show the old value. The updater also fetches the old channel.
@woogerboy21 tracked it down more in detail - taken from gitter chat:
>no matter what i choose always seems to be set back to the value it originally was and i dont see any updates happening to the global config file
which always locks my client into the development branch

>if i manually change the value in the settings file the UI will reflect the change properly. its acting like the change in value in the UI isnt saving the config setting properly

>no matter what i set the value to in the settings dialogue it always checks for updates for the value in the settings file, which never changes even if i change the value in the settings ui under windows

<br>

## Issue 2:
In the info dialog for the `development snapshot` branch, the commit number is not shortened to the usual length:
![development](https://cloud.githubusercontent.com/assets/9874850/23836811/6b9adf6a-077e-11e7-9678-c3099b5be7ac.png)

<br>

## Issue 3:
Also related to that same info dialog, there is a link to the release notes of the last main release.
For development builds that makes not too much sense and a link to the commit on github, or even better the corresponding PR would be more helpful. Not sure what is possible here, maybe there are some other useful information that we could link to there instead...
So basically, we would need a different dialog message for each channel.

Actually, 2 and 3 are described in the PR but ended up different: https://github.com/Cockatrice/Cockatrice/pull/2362#issuecomment-273707379

If I read [this and the following comments](https://github.com/Cockatrice/Cockatrice/pull/2362#issuecomment-282577355) correctly, there is no date included and the updater just checks if the available commit for the selected channel is different from the installed one. 
Is there some way that prevents users from downgrading or otherwise installing a wrong/old/... version with the updater? Right now we are all dependent on bintray or the commits not beeing messed up?

<br>
CC: @ctrlaltca

**Comment by skwerlman** [comment]
For number 3: it's possible to link to a commit range, like this: https://github.com/Cockatrice/Cockatrice/compare/18b8a0ef5c33333a9459900981764338697324cd...f9a025081a8d886da791992411fccb535810b62d

This is how I've seen this done for other projects, like netdata.

**Comment by tooomm** [comment]
>For number 3: it's possible to link to a commit range, like this: 18b8a0e...f9a0250

That's very sweet! Even for several changes it's a nice overview and you can click the linked PR's for more details! 😍
